plugins/emojiextender/EmojiExtenderPlugin.php

```diff
@@ -152,43 +158,59 @@ public function emoji_init_handler($sender, $args) {
      * Configure settings page in dashboard.
      *
      * @param SettingsController $sender
-     * @param array $args
      */
-    public function settingsController_emojiExtender_create($sender, $args) {
+    public function settingsController_emojiExtender_create($sender) {
         $sender->permission('Garden.Settings.Manage');
 
-        $cf = new ConfigurationModule($sender);
         $items = [];
-
         foreach ($this->getEmojiSets() as $key => $emojiSet) {
             $manifest = $this->getManifest($emojiSet);
             $selected = '<svg class="icon icon-svg-checkmark" viewBox="0 0 17 17"><use xlink:href="#checkmark" /></svg>';
-            $icon = (isset($manifest['icon'])) ? img($emojiSet['basePath'].'/'.$manifest['icon'], ['alt' => $manifest['name'], 'class' => 'label-selector-image']) : '';
+            $hasManifestIcon = isset($manifest['icon']);
+            $icon = $hasManifestIcon
+                ? img(
+                    $emojiSet['basePath'] . '/' . $manifest['icon'],
+                    [
+                        'alt' => $manifest['name'],
+                        'class' => 'label-selector-image'
+                    ]
+                )
+                : '';
             $items[$key] =
-                '@<div class="image-wrap">'.
-                $icon.
-                '<div class="overlay">'.
-                   '<div class="buttons">'.
-                        '<a class="btn btn-overlay">'.t('Select').'</a>'.
-                   '</div>'.
-                   '<div class="selected">'.$selected.'</div>'.
-                '</div>'.
-                '</div>'.
-                '<div emojiset-body>'.
-                '<div><b>'.htmlspecialchars($manifest['name']).'</b></div>'.
-                (empty($manifest['author']) ? '' : '<div class="emojiset-author info">'.sprintf(t('by %s'), $manifest['author']).'</div>').
-                (empty($manifest['description']) ? '' : '<p class="emojiset-description">'.Gdn_Format::wysiwyg($manifest['description']).'</p>').
+                '@<div class="image-wrap">' .
+                $icon .
+                '<div class="overlay">' .
+                '<div class="buttons">' .
+                '<a class="btn btn-overlay">' . t('Select') . '</a>' .
+                '</div>' .
+                '<div class="selected">' . $selected . '</div>' .
+                '</div>' .
+                '</div>' .
+                '<div emojiset-body>' .
+                '<div><b>' . htmlspecialchars($manifest['name']) . '</b></div>' .
+                (empty($manifest['author'])
+                    ? ''
+                    : '<div class="emojiset-author info">' . sprintf(t('by %s'), $manifest['author']) . '</div>') .
+                (empty($manifest['description'])
+                    ? ''
+                    : '<p class="emojiset-description">' . Gdn_Format::wysiwyg($manifest['description']) . '</p>') .
```

Comment:
This fragment is a good candidate to be extracted to some small twig template or React component or at least dedicated method.

Reply:
Sure. I was attempting to be more minimal and just update the formatting (pretty sure it only works with Advanced Editor anyways, but I'll give it a go.)

Reply:
I've converted it to a twig view.